### DISCUSSION TITLE
How can I make a certain path seem like a 404

### DISCUSSION THREAD

**Discussion by amirisongithub** [vercel/next.js#67517] [Help] [answered]
### Summary

I need help with something

Let's say for example I have a url that is: amirs.space/supersecrettextfile.txt

now the problem is that I don't want any users actually viewing that txt file

but at the same time, I don't want to just redirect them to 404 page or to the main page, I want it to completely look hidden

the reason why I can't just redirect the users to a 404 page is because vercel's 404 system doesn't redirect to a 404 page, it makes the url, the 404 page
If you don't understand, here's an image for clarification
![image](https://github.com/vercel/next.js/assets/153495399/ff064996-64df-4e61-b284-14e8c1d2db57)

so if I just make that page redirect to the 404.html, the 404 page won't appear at amirs.space/supersecrettextfile.txt, instead it will redirect them to the amirs.space/404.html page and they will notice that supersecrettextfile.txt
(basically, I don't want to just hide the contents of supersecrettextfil.txt, but I also want it to seem like it doesn't exist)

So, how would I hide the supersecrettextfile.txt?

**Comment by NirmalScaria** [comment, marked as answer]
You can handle it with middleware. Perform a [rewrite](https://nextjs.org/docs/app/building-your-application/routing/middleware#nextresponse) for the routes which exist but you want to hide.
